tests/src/JIT/Performance/CodeQuality/Span/Indexer.cs

```diff
@@ -500,9 +500,9 @@ public static void WriteViaIndexer2(int length)
         [MethodImpl(MethodImplOptions.NoInlining)]
         static byte TestWriteViaIndexer2(Span<byte> data)
```

Comment:
This now definitely has the `idx = 0; idx < data.length; idx++` pattern that ought to let us remove the bounds checks from the loop regardless of their relation to one another; I'd suggest making start/length parameters to this like you did with the others.